src/seedsigner/views/tools_views.py

```diff
@@ -241,6 +250,170 @@ def run(self):
+            is_button_text_centered=True,
+            button_data=button_data,
+        ).display()
+
+        if selected_menu_num == RET_CODE__BACK_BUTTON:
+            return Destination(BackStackView)
+
+        elif button_data[selected_menu_num] == TWELVE:
+            return Destination(ToolsCoin_Input_Method_View, view_args=dict(total_flips=mnemonic_generation.COIN__NUM_FLIPS__12WORD))
+
+        elif button_data[selected_menu_num] == TWENTY_FOUR:
+            return Destination(ToolsCoin_Input_Method_View, view_args=dict(total_flips=mnemonic_generation.COIN__NUM_FLIPS__24WORD))
+
+
+
+class ToolsCoin_Input_Method_View(View):
+    def __init__(self, total_flips: int):
+        super().__init__()
+        self.total_flips = total_flips
+
+    def run(self):
+        if self.total_flips == 128:
+            all_flips_text = _("128 coin flips in one go")
+            setwise_text = _("Sets of 11 coin flips")
+        elif self.total_flips == 256:
+            all_flips_text = _("256 coin flips in one go")
+            setwise_text = _("Sets of 11 coin flips")
+        else:
+            raise ValueError("Unsupported flip count")
+
+        ALL_FLIPS = ButtonOption(all_flips_text, return_data="all")
+        SETWISE = ButtonOption(setwise_text, return_data="setwise")
+
+        button_data = [ALL_FLIPS, SETWISE]
+        selected_menu_num = ButtonListScreen(
+            title=_("Input Method"),
+            is_bottom_list=True,
+            is_button_text_centered=True,
+            button_data=button_data,
+        ).display()
+
+        if selected_menu_num == RET_CODE__BACK_BUTTON:
+            return Destination(ToolsCoinEntropyMnemonicLengthView)
+
+        selected_option = button_data[selected_menu_num].return_data
+
+        if selected_option == "all":
+            return Destination(ToolsCoinEntropyEntryView, view_args={"total_flips": self.total_flips})
+
+        elif selected_option == "setwise":
+            return Destination(ToolsCoinEntropySetwiseEntryView, view_args={"total_flips": self.total_flips})
+
+
+
+class ToolsCoinEntropyEntryView(View):
+    def __init__(self, total_flips: int):
+        super().__init__()
+        self.total_flips = total_flips
+    
+
+    def run(self):
+        ret = ToolsCoinEntropyEntryScreen(
+            return_after_n_chars=self.total_flips,
+        ).display()
+
+        if ret == RET_CODE__BACK_BUTTON:
+            return Destination(BackStackView)
+
+        coin_seed_phrase = mnemonic_generation.generate_mnemonic_from_coin_flips(ret)
+
+        # Add the mnemonic as an in-memory Seed
+        seed = Seed(coin_seed_phrase, wordlist_language_code=self.settings.get_value(SettingsConstants.SETTING__WORDLIST_LANGUAGE))
+        self.controller.storage.set_pending_seed(seed)
+
+        # Cannot return BACK to this View
+        return Destination(SeedWordsWarningView, view_args={"seed_num": None}, clear_history=True)
+
+
+
+class ToolsCoinEntropySetwiseEntryView(View):
```

Comment:
It's more obvious from the name, but still a good idea to have an explanatory comment for the `View`.